### PR TITLE
Dual quaternion

### PR DESCRIPTION
**Pull request by beeshi**

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -10,14 +10,16 @@
...
                                # slerp, squad,
                                )
+from .numpy_dual_quaternion import dual_quaternion
+
 from .quaternion_time_series import slerp, squad, integrate_angular_velocity, minimal_rotation
 from .calculus import derivative, definite_integral, indefinite_integral
 from ._version import __version__
 
 __doc_title__ = "Quaternion dtype for NumPy"
 __doc__ = "Adds a quaternion dtype to NumPy."
```
Comment (igilitschenski) [inline __init__.py]: Maybe adapt this documentation....

File: __init__.py [hunk 2]
```diff
@@ -10,14 +10,16 @@
...
 
-__all__ = ['quaternion',
+__all__ = ['quaternion', 'dual_quaternion',
```
Comment (igilitschenski) [inline __init__.py]: Does `as_dual_quat_array` need to be exported here as well?

File: numpy_dual_quaternion.c [hunk 3]
```diff
@@ -0,0 +1,1436 @@
...
+  int dual_quaternionNum;
+  int arg_types[3];
+  PyObject* numpy;
+  PyObject* numpy_dict;
+
+  // Initialize a (for now, empty) module
+#if PY_MAJOR_VERSION >= 3
+  module = PyModule_Create(&moduledef);
+#else
+  module = Py_InitModule("numpy_dual_quaternion", DualQuaternionMethods);
```
Comment (igilitschenski) [inline numpy_dual_quaternion.c]: The identifier `DualQuaternionMethods` has never been declared, which results in not-compiling on my system. Have you been testing with python3 (which maybe does not have this problem)?

**Review by igilitschenski — COMMENTED** [review]
Left some first comments. Only very few for now as my system ran into compile issues which may be due to the fact that we used different python versions. Could you also try with Python2 or did you use it already?

File: numpy_dual_quaternion.c (hunk 3 above)
Comment (beeshi) [reply numpy_dual_quaternion.c]: Yeah I've been using python3 - I'll try testing with python2 now

**Comment by igilitschenski** [comment]
I tried everything out. Will merge this now and open a PR into @moble's original repo. Then, we can add more features later.

**Pull request merged**